### DISCUSSION TITLE
how to use enum with multiple values in Assert\Choice ?

### DISCUSSION THREAD

**Discussion by kconde2** [symfony/symfony#47691] [Q&A]
```php
enum RoleEnum: string
{
    case ADMIN = 'ROLE_ADMIN';
    case USER = 'ROLE_USER';
}
```

```php
class User {
    #[Assert\Choice(choices: RoleEnum::cases(), multiple: true)] // <- this is not working
    private array $roles = [];
}
```

I have this error message
![image](https://user-images.githubusercontent.com/43377621/192220427-7f7e6140-7fa6-40d2-8591-62837c776167.png)

Environnement
Symfony 6.1.4
php 8.1.10
Thanks in advance for helping me

**Comment by MatTheCat** [comment]
```php
use Symfony\Component\Validator\Constraints\Type;

class User {
    #[All(new Type(RoleEnum::class))]
    private array $roles = [];
}
```
should do the trick.

**Reply by kconde2** [reply]
I got this
![image](https://user-images.githubusercontent.com/43377621/192369609-a35ffe7d-0ab7-4bef-8bbf-3f3501426513.png)

**Reply by kconde2** [reply]
It's not working even with correct roles. Here I'm using Api Platform
![image](https://user-images.githubusercontent.com/43377621/192369686-0c606b6f-93ec-41ee-972d-7a9c33bdbae0.png)

**Reply by MatTheCat** [reply]
Well `RoleEnum::cases` returns instances of `RoleEnum` but you're validating a string...

See https://github.com/symfony/symfony/pull/43047 for a discussion and solutions about your use-case.

**Reply by kconde2** [reply]
I will see there.

Thanks for your help.

**Reply by kconde2** [reply]
@MatTheCat 

I finally did this 

```php
 #[Assert\Choice(callback: [RoleEnum::class, 'values'] , multiple: true')]
```

And inside `RoleEnum.php`

```php
 public static function values(): array
 {
     return array_column(self::cases(), 'value');
 }
```

Reference here : https://github.com/symfony/symfony/pull/43047#issuecomment-1161704955

Thanks again for your help.
